Setuptools removed distutils.msvccompiler which breaks the NumPy build.

Not intended for upstream, they already resolved this in NumPy 2.x.

--- numpy-1.26.4/numpy/core/setup.py
+++ numpy-1.26.4/numpy/core/setup.py
@@ -9,7 +9,6 @@ import glob
 from os.path import join
 
 from numpy.distutils import log
-from numpy.distutils.msvccompiler import lib_opts_if_msvc
 from distutils.dep_util import newer
 from sysconfig import get_config_var
 from numpy.compat import npy_load_module
@@ -688,7 +687,7 @@ def configuration(parent_package='',top_
             install_dir='lib',
             build_info={
                 'include_dirs' : [],  # empty list required for creating npy_math_internal.h
-                'extra_compiler_args': [lib_opts_if_msvc],
+                'extra_compiler_args': [],
             })
     config.add_npy_pkg_config("npymath.ini.in", "lib/npy-pkg-config",
             subst_dict)
--- numpy-1.26.4/numpy/random/setup.py
+++ numpy-1.26.4/numpy/random/setup.py
@@ -3,7 +3,6 @@ import sys
 from os.path import join
 
 from numpy.distutils.system_info import platform_bits
-from numpy.distutils.msvccompiler import lib_opts_if_msvc
 
 
 def configuration(parent_package='', top_path=None):
@@ -70,7 +69,7 @@ def configuration(parent_package='', top
         not initialized the distutils build command, so use this deferred
         calculation to run when we are building the library.
         """
-        opts = lib_opts_if_msvc(build_cmd)
+        opts = []
         if build_cmd.compiler.compiler_type != 'msvc':
             # Some bit generators require c99
             opts.append('-std=c99')
